syntax = "proto3";

package RFIT_NS;

message Message {
  uint32 id = 1;

  string funcName = 2;

  uint64 timestamp = 3;
  uint64 finishTimestamp = 4;

  bytes inputData = 5;
  bytes outputData = 6;

  bool isPing = 7;
}

message FunctionRegisterMsg{
  string funcName = 1;

  uint64 memSize = 2;
  double coreRation = 3;
  uint32 concurrency = 4;

  bytes dlData = 5;

  bytes type = 6;

  // ResponseInfo
  bool status = 7;
  string message = 8;
}